vaultpress.php

```diff
@@ -9,18 +9,69 @@
  * License: GPL2+
  * Text Domain: vaultpress
  * Domain Path: /languages/
+ *
+ * @package VaultPress
+ */
+
+// don't call the file directly.
+defined( 'ABSPATH' ) || die();
+
+define( 'VAULTPRESS__MINIMUM_PHP_VERSION', '5.3.2' );
+define( 'VAULTPRESS__VERSION', '2.0-beta' );
+define( 'VAULTPRESS__PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
+
+/**
+ * First, we check for our supported version of PHP. If it fails,
+ * we "pause" VaultPress by ending the loading process and displaying an admin_notice to inform the site owner.
  */
+if ( version_compare( phpversion(), VAULTPRESS__MINIMUM_PHP_VERSION, '<' ) ) {
+	if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+		error_log(
+			sprintf(
+				/* translators: Placeholders are numbers, versions of PHP in use on the site, and required by VaultPress. */
+				esc_html__( 'Your version of PHP (%1$s) is lower than the version required by VaultPress (%2$s). Please update PHP to continue enjoying VaultPress.', 'vaultpress' ),
+				esc_html( phpversion() ),
+				VAULTPRESS__MINIMUM_PHP_VERSION
+			)
+		);
+	}
+
+	/**
+	 * Outputs an admin notice for folks running an outdated version of PHP.
+	 *
+	 * @todo: Remove once WP 5.2 is the minimum version.
+	 *
+	 * @since 2.0.0
+	 */
+	function vaultpress_admin_unsupported_php_notice() { ?>
+		<div class="notice notice-error is-dismissible">
+			<p><?php esc_html_e( 'VaultPress requires a more recent version of PHP and has been paused. Please update PHP to continue enjoying VaultPress.', 'vaultpress' ); ?></p>
+			<p class="button-container">
+				<?php
+				printf(
+					'<a class="button button-primary" href="%1$s" target="_blank" rel="noopener noreferrer">%2$s <span class="screen-reader-text">%3$s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a>',
+					esc_url( wp_get_update_php_url() ),
```

Comment:
Could `wp_get_update_php_url()` fatal a pre-5.1 WP install?

Reply:
Yes, that's a good point. I assume we don't want to have VaultPress require WP 5.1 just yet.

b2b2a71d5119d24cdf8a08d6f56a24be0dc96c15 should solve this issue.